js_modules/dagster-ui/packages/ui-core/src/runs/useRunsForTimeline.tsx

```diff
@@ -41,16 +41,12 @@ export const useRunsForTimeline = (
   const startSec = start / 1000.0;
   const endSec = end / 1000.0;
 
-  const buckets = useMemo(() => {
-    const buckets = [];
-    for (let time = start; time < end; time += BUCKET_SIZE_MS) {
-      buckets.push([time, Math.min(end, time + BUCKET_SIZE_MS)] as const);
-    }
-    return buckets;
-  }, [start, end]);
+  const buckets = useMemo(() => getHourlyBuckets(start, end), [start, end]);
 
   const client = useApolloClient();
 
+  const completedRunsCache = useMemo(() => new HourlyDataCache<RunTimelineFragment>(), []);
```

Comment:
Hmm it doesn't seem like the cache is filter-aware, I think that when the `runsFilter` passed to this hook changes this cache needs to be replaced?

Reply:
Ah yeah, the current use case doesn't actually use filtering, does the new executions page you're working on need filtering?